src/rules/arrowParensRule.ts

```diff
@@ -72,8 +72,10 @@ function walk(ctx: Lint.WalkContext<Options>) {
                 }
             } else if (ctx.options.banSingleArgParens) {
                 const closeParen = getChildOfKind(node, ts.SyntaxKind.CloseParenToken)!;
+                const charBeforeOpenParen = ctx.sourceFile.text.substring(openParen.pos - 1, openParen.pos);
+                const replaceValue = charBeforeOpenParen.match(/[a-z=]/i) !== null ? " " : "";
```

Comment:
I don't think this rule should do anything related to code formatting. It adds the space only where strictly necessary to avoid invalid syntax.
Everything else assumes a certain style preference of the user, that this rule cannot know.

Anyways, if you really want to add a space after `=` you should probably also add one after
* `,` as in `foo(bar, (a) => b)`
* `?` and `:` as in `foo ? (a) => b : (a) => c`
* `&` and `|` as in `someFn || (a) => b`
* maybe more

Reply:
I think you're right about unnecessary code formatting here. I'll change it and adjust test cases accordingly.